### DISCUSSION TITLE
Support for international characters?

### DISCUSSION THREAD

**Discussion by andrewbeno** [dfinke/ImportExcel#1606] [General]
I tried searching discussions for this type of topic, but haven't come across any so I figured I'd post here.

I'm importing a CSV file that has some characters with accents, i.e., "Br**é**sil", and then using Export-Excel to generate a XLSX file, but the resulting XLSX file doesn't "preserve" them.  Is there a way to workaround that?

Here's a portion of the script:
```
$data = Import-Csv -Path $destinationPath -Encoding UTF8
$ExcelFilename = (Get-Item $destinationPath).Basename + ".xlsx"
$data | Export-Excel .\$ExcelFilename -AutoNameRange -Show
```

Here's what one line of the CSV file looks like:
![image](https://github.com/dfinke/ImportExcel/assets/58045340/1725e570-876b-4c89-847d-fb0ee2b91333)

Here's what that same line in the exported XLSX file looks like:
![image](https://github.com/dfinke/ImportExcel/assets/58045340/1ac51a72-ce68-472a-ac9a-be0ada7a55d0)

Thanks for any input/advice!

Andy

**Comment by dfinke** [comment]
@scriptingstudio thoughts?

**Reply by scriptingstudio** [reply]
### My op environment
Windows 11 23H2 En with Russian support
MSO 2021
VS Code
Windows Terminal
Powershell 5.1 / 7.4.2

```powershell
get-culture

LCID             Name             DisplayName
----             ----             -----------
1049             ru-RU            Russian (Russia)
```

### Steps to reproduce
Script:
```powershell
& {
	$data = Import-Csv test1.csv -Encoding utf8 -Delimiter ';'
	$ExcelFilename = 'test1.xlsx'
	if (Test-Path $ExcelFilename) {Remove-Item $ExcelFilename}
	$data | Export-Excel $ExcelFilename -AutoNameRange -Show
}
```

Datafile:
```
Id;Name;Lang
support@;Brésil;France|Французский/Бразильский
support@;Нью-Йорк;Russian|Русский
support@;巴西;China|Китай
```

CSV output:
```powershell
import-csv test1.csv -delim ';'

Id       Name     Lang
--       ----     ----
support@ Brésil   France|Французский/Бразильский
support@ Нью-Йорк Russian|Русский
support@ 巴西     China|Китай
```

I tested data with both encodings (UTF8BOM and UTF8noBOM) in both versions of PowerShell - 4 cases.

Input & output files:
[test1.csv](https://github.com/dfinke/ImportExcel/files/15441992/test1.csv)

[test1.xlsx](https://github.com/dfinke/ImportExcel/files/15441993/test1.xlsx)

Data file in UTF8BOM opened in Excel looks like this:
![image](https://github.com/dfinke/ImportExcel/assets/17237559/35dc5238-6872-44a7-831a-97f68e9171a9)

Data file in UTF8noBOM opened in Excel looks like this:
![image](https://github.com/dfinke/ImportExcel/assets/17237559/8ef321f8-3a4f-44e7-8548-af9e39dae2b4)


I had no any issue with results. It should just work. Datafile encoding does not matter (BOM or NOBOM), output is always correct.
**I think the problem is in @andrewbeno environment or CSV file has wrong encoding (non UTF)**.

### Additional test with non-UTF encoding
Encoding is Windows 1251
output file:
![image](https://github.com/dfinke/ImportExcel/assets/17237559/e3460cec-af8d-4157-b5f4-fd9c4247d17d)

I'm inclined to think that the issue is non-UTF encoding of datafile (most probably ASCII).
Symbol 
![image](https://github.com/dfinke/ImportExcel/assets/17237559/7312ddea-17c6-4abb-b220-59c4ee211c01)
means lost data

I played with different encodings but no luck.
Cannot say anymore without seeing real datafile.

**Reply by dfinke** [reply]
Thanks, super!

**Comment by scriptingstudio** [comment]
![image](https://github.com/dfinke/ImportExcel/assets/17237559/f29843b1-5c79-4da2-be44-72d50fcc7bfc)
Question mark implies lost data which can mean corrupted source or unsuitable argument of `-encoding` parameter for `import-csv`.
The ImportExcel module does not do anything with encoding.

**Reply by andrewbeno** [reply]
Thanks @dfinke and @scriptingstudio!  I should've done more troubleshooting before posting, but I appreciate the help and hope you didn't spend too much time investigating.  After reading your response, I read that one way to confirm the encoding of a CSV file is in Notepad++ in the lower-right corner, and saw that the CSV I was importing was not encoded with UTF-8 as you suggested.
![image](https://github.com/dfinke/ImportExcel/assets/58045340/ac2e208d-df34-46d9-b7f7-676f8e8a2b6f)

I was able to workaround the issue by using the 'Default' encoding (initially I had no encoding parameter and then only tested with UTF8 before posting.. can't believe I didn't try "Default" :expressionless: ) 
```
$data = Import-Csv -Path $destinationPath -Encoding Default
$ExcelFilename = (Get-Item $destinationPath).Basename + ".xlsx"
$data | Export-Excel .\$ExcelFilename -AutoNameRange -Show
```

Thanks again and have a good one!
Andy

**Comment by dfinke** [comment]
Thanks @andrewbeno. Did not know `-Encoding` existed on `Import-Csv`
